### ISSUE TITLE
Webview2 Installer Restarts PC Without Warning

### ISSUE THREAD

**Issue by conradakunga** [RickStrahl/MarkdownMonster#853]
OS=Windows 10.0.19043.1165 (21H1/May2021Update)
Intel Core i5-9300H CPU 2.40GHz, 1 CPU, 8 logical and 4 physical cores
.NET SDK=6.0.100-preview.7.21379.14

I use the chocolatey installer and I have noticed a couple of things

1. The Webview2 is installed every single time I run `choco upgrade markdownmonster` to upgrade
2. After the webview2 installs the machine is unceremoniously restarted

This is the third time I think this has happened with MM 2

The former is annoying but bearable. The latter is very disruptive.

Is this something you have any control over?

**Comment by RickStrahl** [comment]
Yeah that sounds annoying but that's not what I'm seeing here on my setups. Yes the WebView installs each time, but it never reboots my machine - I install MM multiple times a day and on several machines so I'm pretty sure this isn't standard behavior.

Reinstalls typically happen when files are locked. I have the option to force a reinstall from the installer turned off, so that part never trigger a restart. It must be the WebView install itself that's causing this to happen.

Lets see if anybody else is running into this.

**Comment by conradakunga** [comment]
So today I ran the update for 2.0.14 to 2.0.15

I had also thought the problem may have been due to locked files - Edge perhaps.

No restart this time, but I made sure that there were a bunch of Edge instances running prior to the upgrade.

Next time I'll try upgrading with MM itself open and see what happens.

**Comment by sharonxmcd** [comment]
Hi there, I'm very new to this & I was wondering if MM can be used on a
mobile phone or does it need to be a laptop or PC? Also, I was hoping you
could give a short explanation of the basics of MM, what is it best used
for? Deepfakes too what devices can you use to make Deepfakes? Any
information & advice would be greatly appreciated.

Best wishes,

Sharon.

**Comment by RickStrahl** [comment]
If MM is running and you're doing a silent install (via Chocolatey) I think that may be the problem. If you run an interactive install the installer prompts for unloading the app, but I believe in a silent install that doesn't happen. 

FWIW, It's never a good idea to install an app that is running... :smile:

**Comment by conradakunga** [comment]
So I got the restart on upgrading from 15 to 16.

As far as I can tell I had pretty much the same apps open when upgrading from 14 to 15.

Next time will take a process dump of what is running

**Comment by RickStrahl** [comment]
The only thing I can think of is that there is another app running that has the WebView loaded.

MM gets auto shut down during the install so that shouldn't be the issue.

**Comment by conradakunga** [comment]
So the upgrade from 17 to 18 unceremoniously restarted.

I took care to take a dump of the running processes. I have attached. Nothing seems to jump out.

[process.txt](https://github.com/RickStrahl/MarkdownMonster/files/7291225/process.txt)

Is it possible to detect the presence of thee WebView during install and skip the re-installation? Or is there a particular reason it is always re-installed?

**Comment by RickStrahl** [comment]
The problem is that the WebView runtime has to be updated to stay current - we always want the latest runtime installed, so while I can check for installation, I can't determine whether it needs updating inside of my installer. MM updates it's WebView libraries whenever they come out to ensure we more or less stay in sync with the published runtimes.

Unfortunately Microsoft doesn't check if the runtimes are up to do date or not and just reinstalls the WebView runtimes every time - that sucks, but there's no way I know of to change that behavior. The runtime update roughly monthly and are in line with Chromium updates.

As said - I've **never** seen a reboot from a WebView runtime install, so there's something funky going on on your system I think that has to do with the runtimes being locked or active in memory. Perhaps another app running that's also using the runtimes. MM should get shut down if it's running, so that's likely not it.
